### PR TITLE
[tune] Fix trial cleanup after x seconds, set default to 600

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

This currently does not work in three places: 1) We need to kill the actor as garbage collection will not work with futures in flight, 2) We need to trigger the _stop_actor method after clearing the futures, as it will create a new future, 3) the future was not fetched correctly.

We also set the default cleanup time to 10 minutes, which should suffice for most cases and avoids deadlocks in long-running tasks.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/execution/ray_trial_executor.py [hunk 1]
```diff
@@ -85,12 +85,14 @@ def unwrap(self):
...
 
-def _post_stop_cleanup(future, pg):
+def _post_stop_cleanup(future, actor, pg):
     """Things to be done after a trial is stopped."""
+    assert isinstance(actor, ray.actor.ActorHandle)
     assert isinstance(pg, PlacementGroup)
     try:
         # This should not be blocking as
         # we are only here when triggered.
+        ray.kill(actor)
```
Comment (xwjiang2010) [inline python/ray/tune/execution/ray_trial_executor.py:95]: does delete work in this case? Just curious..

File: python/ray/tune/trainable/trainable.py [hunk 2]
```diff
@@ -1131,7 +1131,7 @@ def cleanup(self):
         If any Ray actors are launched in the Trainable (i.e., with a RLlib
         trainer), be sure to kill the Ray actor process here.
 
-        You can kill a Ray actor by calling `actor.__ray_terminate__.remote()`
+        You can kill a Ray actor by calling `ray.kill(actor)`
```
Comment (xwjiang2010) [inline python/ray/tune/trainable/trainable.py:1134]: can we also mention that this method is supposed to be light-weight. Also the forced clean up will kick in after a default of 600 seconds (maybe pointing them to the env doc string).

**Review by xwjiang2010 — APPROVED** [review]
Thanks Kai for fixing this!

**Review by Yard1 — APPROVED** [review]
Looks great!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**